### PR TITLE
Update cost optimized state to use pydbapi_extracted functionality

### PR DESCRIPTION
**Pull request by Simranpal**
With this PR the cost optimized state will use our custom `pydbapi_extracted` function instead of salt's `archive.extracted` to utilize the automatic detection / finding hana software folders within sap media

Also removed the hardcoded client package path and now it uses the software path for pyadbapi location

To-Do:
-Add support for HANA 1.0 Client which has different folder structure. It has the client files inside `hdcli` folder inside the `pydbapi` tar vs `hdcli-version-tar`
-Test with deployment scenarios
-Update spec, change log

Related to https://github.com/SUSE/salt-shaptools/pull/61

### CODE REVIEW

File: hana/enable_cost_optimized.sls [hunk 1]
```diff
@@ -50,15 +50,14 @@ failure:
...
-
 extract_hana_pydbapi_archive:
-    archive.extracted:
-      - name: /hana/shared/srHook
-      - enforce_toplevel: False
-      - source: {{ py_packages_folder }}
-      - options: --transform 's|-[0-9]*\.[0-9]*\.[0-9]*|-package|' --wildcards 'hdbcli*'
+    hana.pydbapi_extracted:
+      - name: PYDBAPI.TGZ
+      - software_folders: [{{ hana.software_path }}]
```
Comment (arbulu89) [inline hana/enable_cost_optimized.sls:56]: I think we need to put `node.install.software_path|default(hana.software_path)` here

Comment (Simranpal) [reply hana/enable_cost_optimized.sls]: Thanks, I updated it

**Comment by arbulu89** [comment]
@Simranpal I think we can postpone the `HANA 1.0` implementation by now. What do you think?
Eventually, if `HANA 1.0` is required (we need to discuss this), we will do all the needed changes. I think there are more than only this.

Besides that, we only need to do the other things you wrote in the main description.

**Comment by Simranpal** [comment]
Thanks @arbulu89 as discussed this works for `HANA 2.0` I did the testing with cost opt deployment and works as expected

**Comment by arbulu89** [comment]
> Thanks @arbulu89 as discussed this works for `HANA 2.0` I did the testing with cost opt deployment and works as expected

Thanks @Simranpal , so update the spec and changes file and we can merge

**Pull request merged**